src/charm.py

```diff
@@ -176,22 +180,34 @@ def _set_alerts(self, container):
                 alert rule files need to be created.
         """
         if not self.prometheus_ready:
+            logger.debug("Abort processing alert rules: prometheus not ready")
             return
 
         with container.is_ready():
+            logger.debug("Processing alert rules")
+
             container.remove_path(RULES_DIR, recursive=True)
 
-        for rel_id, alerts in self.metrics_consumer.alerts().items():
-            filename = "juju_{}_{}_{}_rel_{}_alert.rules".format(
-                alerts["model"], alerts["model_uuid"], alerts["application"], rel_id
-            )
-            try:
-                path = os.path.join(RULES_DIR, filename)
-                rules = yaml.dump({"groups": alerts["groups"]})
-                logger.debug("Rules for relation %s : %s", rel_id, rules)
-                container.push(path, rules, make_dirs=True)
-            except ConnectionError:
-                logger.error("Failed to write alert rule %s", filename)
+            alert_rules_by_rel_id = self.prometheus_provider.alerts()
+
+            if alert_rules_by_rel_id:
```

Comment:
This if condition is not necessary. `alerts()` returns and empty dictionary if no alerts are available. Hence the for loop iteration will always work but in the case of an empty dictionary no iterations will occur (same as if condition).